packages/ccda/src/systems.ts

```diff
@@ -0,0 +1,630 @@
+ * Map the C-CDA system to the FHIR system.
+ * @param ccda - The C-CDA system to map.
+ * @returns The FHIR system.
+ */
+export function mapCcdaSystemToFhir(ccda: string | undefined): string | undefined {
+  if (!ccda) {
+    return undefined;
+  }
+  return SYSTEM_MAPPER.mapCcdaToFhir(ccda) ?? `urn:oid:${ccda}`;
+}
+
+/**
+ * Map the FHIR system to the C-CDA system.
+ * @param system - The system to map.
+ * @returns The C-CDA system.
+ */
+export function mapFhirSystemToCcda(system: string | undefined): string | undefined {
+  if (!system) {
+    return undefined;
+  }
+  if (system.startsWith('urn:oid:')) {
+    return system.replace('urn:oid:', '');
+  }
+  return SYSTEM_MAPPER.mapFhirToCcda(system);
+}
+
+/**
+ * Map the codeable concept to the C-CDA code.
+ * @param codeableConcept - The codeable concept to map.
+ * @returns The C-CDA code.
+ */
+export function mapCodeableConceptToCcdaCode(codeableConcept: CodeableConcept | undefined): CcdaCode | undefined {
+  if (!codeableConcept) {
+    return undefined;
+  }
+
+  const systemEntry = codeableConcept.coding?.[0]?.system
+    ? SYSTEM_MAPPER.getEntryByFhir(codeableConcept.coding[0].system)
+    : undefined;
+  const system = systemEntry?.ccdaValue;
+  const systemName = systemEntry?.displayName;
+
+  return {
+    '@_code': codeableConcept?.coding?.[0]?.code,
+    '@_displayName': codeableConcept?.coding?.[0]?.display,
+    '@_codeSystem': system,
+    '@_codeSystemName': systemName,
+  };
+}
+
+/**
+ * Map the codeable concept to the C-CDA value.
+ * @param codeableConcept - The codeable concept to map.
+ * @returns The C-CDA value.
+ */
+export function mapCodeableConceptToCcdaValue(codeableConcept: CodeableConcept | undefined): CcdaValue | undefined {
+  const code = mapCodeableConceptToCcdaCode(codeableConcept);
+  if (!code) {
+    return undefined;
+  }
+  return {
+    '@_xsi:type': 'CD',
+    ...code,
+  };
+}
+
+export const CONFIDENTIALITY_MAPPER = new EnumMapper(
+  'Confidentiality',
+  OID_CONFIDENTIALITY_VALUE_SET,
+  CONFIDENTIALITY_CODE_SYSTEM,
+  [
+    { ccdaValue: 'U', fhirValue: 'U', displayName: 'unrestricted' },
+    { ccdaValue: 'L', fhirValue: 'L', displayName: 'low' },
+    { ccdaValue: 'M', fhirValue: 'M', displayName: 'moderate' },
+    { ccdaValue: 'N', fhirValue: 'N', displayName: 'normal' },
+    { ccdaValue: 'R', fhirValue: 'R', displayName: 'restricted' },
+    { ccdaValue: 'V', fhirValue: 'V', displayName: 'very restricted' },
+  ]
+);
+
+export const HUMAN_NAME_USE_MAPPER = new EnumMapper('HumanNameUse', '', NAME_USE_VALUE_SET, [
+  { ccdaValue: 'C', fhirValue: 'usual', displayName: 'Common/Called by' },
+  { ccdaValue: 'L', fhirValue: 'official', displayName: 'Legal' },
+  { ccdaValue: 'TEMP', fhirValue: 'temp', displayName: 'Temporary' },
+  { ccdaValue: 'N', fhirValue: 'nickname', displayName: 'Nickname' },
+  { ccdaValue: 'ANON', fhirValue: 'anonymous', displayName: 'Anonymous' },
+  { ccdaValue: 'M', fhirValue: 'maiden', displayName: 'Maiden' },
```

Comment:
Note: Entries with same `ccdaValue` and different `fhirValue`s map given `ccdaValue` to the first `fhirValue` always, but will map any of the matching `fhirValue`s to the one `ccdaValue`.

Might be worth notating somewhere with a comment